File: src/dataset/cpp/isogram_Exercism_thefullarcticfox/claude/ClaudeSonnet4_isogram_v4.cpp
#include "isogram.h"
#include <unordered_set>

namespace isogram {
	bool	is_isogram(const std::string& s) {
		if (s.empty())
			return true;

		std::unordered_set<char> seen;
		seen.reserve(26);

		for (char c : s) {
			if (c == ' ' || c == '-')
				continue;
			
			char lower_c = (c >= 'A' && c <= 'Z') ? c + 32 : c;
			
			if (!seen.insert(lower_c).second)
				return false;
		}
		return true;
	}
}  // namespace isogram

File: src/dataset/javascript/isogram_exercism-javascript-bearguns/claude/ClaudeSonnet4_isogram_v4.js
var Isogram = function(string) {
  this.string = string;
  this.isIsogram = function() {
    var cleaned = this.string.toLowerCase().replace(/[\s-,\.]/g,'');
    var seen = new Set();
    
    for (var i = 0; i < cleaned.length; i++) {
      var char = cleaned[i];
      if (seen.has(char)) {
        return false;
      }
      seen.add(char);
    }
    
    return true;
  }
};

export default Isogram;